src/components/UserList.jsx

```diff
@@ -1,4 +1,42 @@
+import { useContext } from "react";
+import userContext from "../store/userContext";
+import SyncLoader from "react-spinners/SyncLoader";
+
 const UserList = () => {
-  return <div>UserList</div>;
+  const { users, loginUser, setLoginUser, isLoading } = useContext(userContext);
+
+  return (
+    <>
+      {isLoading ? (
+        <div className="overlay">
+          <SyncLoader className="spinner" />
+        </div>
```

Comment:
If you're re-using this Loading div in more than one place it might be cool to extract this into a component too